google-cloud-storage/src/main/java/com/google/cloud/storage/DefaultStorageWriterConfig.java

```diff
@@ -0,0 +1,125 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.storage;
+
+import com.google.api.core.ApiFuture;
+import com.google.api.core.ApiFutures;
+import com.google.api.core.BetaApi;
+import com.google.api.core.InternalApi;
+import com.google.cloud.storage.BufferedWritableByteChannelSession.BufferedWritableByteChannel;
+import com.google.cloud.storage.Conversions.Decoder;
+import com.google.cloud.storage.UnifiedOpts.ObjectTargetOpt;
+import com.google.cloud.storage.UnifiedOpts.Opts;
+import com.google.common.base.Preconditions;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.google.storage.v2.WriteObjectResponse;
+import java.nio.channels.WritableByteChannel;
+import java.time.Clock;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Configure a writer which is logically equivalent to the following:
+ *
+ * <pre>{@code
+ * Storage storage = ...;
+ * WriteChannel writeChannel = storage.writer(BlobInfo, BlobWriteOption);
+ * writeChannel.setChunkSize(chunkSize);
+ * }</pre>
+ */
+@Immutable
+@BetaApi
+public final class DefaultStorageWriterConfig extends StorageWriterConfig {
+
+  private final int chunkSize;
+
+  @InternalApi
+  DefaultStorageWriterConfig(int chunkSize) {
+    this.chunkSize = chunkSize;
+  }
+
+  public int getChunkSize() {
+    return chunkSize;
+  }
+
+  @BetaApi
+  public DefaultStorageWriterConfig withChunkSize(int chunkSize) {
+    Preconditions.checkArgument(
+        chunkSize >= ByteSizeConstants._256KiB,
+        "chunkSize must be >= %d",
+        ByteSizeConstants._256KiB);
+    return new DefaultStorageWriterConfig(chunkSize);
+  }
+
+  @Override
+  @InternalApi
+  WriterFactory createFactory(Clock clock) {
+    return new Factory(chunkSize);
+  }
+
+  @InternalApi
+  private static final class Factory implements WriterFactory {
+
+    private final int chunkSize;
+
+    private Factory(int chunkSize) {
+      this.chunkSize = chunkSize;
+    }
+
+    @InternalApi
+    @Override
+    public WritableByteChannelSession<?, BlobInfo> writeSession(
+        StorageInternal s,
+        BlobInfo info,
+        Opts<ObjectTargetOpt> opts,
+        Decoder<WriteObjectResponse, BlobInfo> d) {
+      // todo: invert this
+      //   make GrpcBlobWriteChannel use this factory to produce its WriteSession
+      if (s instanceof GrpcStorageImpl) {
+        GrpcStorageImpl g = (GrpcStorageImpl) s;
+        GrpcBlobWriteChannel writer = g.writer(info);
+        writer.setChunkSize(chunkSize);
+        WritableByteChannelSession<BufferedWritableByteChannel, WriteObjectResponse> session =
+            writer.newLazyWriteChannel().getSession();
+        return new DecoratedWritableByteChannelSession<>(session, d);
+      }
+      return CrossTransportUtils.throwGrpcOnly(DefaultStorageWriterConfig.class, "");
```

Comment:
No. This is an internal only method that will never be on the public api. This error fall through is to catch if somehow someone does manage to call this in a way they shouldn't it will error.